hack/run-local.sh

```diff
@@ -14,5 +14,11 @@ echo "Image: ${IMAGE}"
 echo "Release version: ${RELEASE_VERSION}"
 echo "Namespace: ${NAMESPACE}"
 
-${DELVE:-} ./ingress-operator start --image "${IMAGE}" --release-version "${RELEASE_VERSION}" \
+if [[ -v "ENABLE_CANARY" ]]; then
+    CANARY_ARG="--canary-image"
+    CANARY_IMAGE=$(oc get -n openshift-ingress-operator deployments/ingress-operator -o json | jq -r '.spec.template.spec.containers[0].env[] | select(.name=="CANARY_IMAGE").value')
+    echo "Canary Image: ${CANARY_IMAGE}"
+fi
+
+${DELVE:-} ./ingress-operator start --image "${IMAGE}" ${CANARY_ARG:-} ${CANARY_IMAGE:-} --release-version "${RELEASE_VERSION}" \
```

Comment:
This logic breaks if the user specifies `CANARY_IMAGE` directly.  You could use `${CANARY_IMAGE:+--canary-image=${CANARY_IMAGE}}`, or even simpler, `--canary-image=${CANARY_IMAGE:-}` (the equals sign ensures the subsequent parameter is not used as the value, so the empty string is used as the parameter value for `--canary-image`).  

I don't personally use this shell script though, so don't waste time on this if you are happy with it as you have it.

Reply:
I have updated the shell script based off of your comments.